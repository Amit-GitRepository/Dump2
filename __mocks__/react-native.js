var rn = require('react-native');

rn.NativeModules.RNI18n = {languages: ['en']};
rn.NativeModules.BlobModule = {BLOB_URI_SCHEME: ''};
rn.NativeModules.FirebaseDynamicLink = {
  addListener: jest.fn(),
  removeListeners: jest.fn()
};
rn.NativeModules.ReactNativeExceptionHandler = {
  setHandlerforNativeException: jest.fn()
};
jest.mock('AsyncStorage', () => ({
  getItem: jest.fn(() => Promise.resolve('{}')),
  setItem: jest.fn(() => Promise.resolve()),
  mergeItem: jest.fn(() => Promise.resolve()),
  multiGet: jest.fn(() => Promise.resolve('{}'))
}));
jest.mock('BackHandler', () => ({
  addEventListener: jest.fn()
}));
jest.mock('AppState', () => ({
  currentState: 'inactive',
  addEventListener: jest.fn()
}));
var ReactNativeART = require('react-native/Libraries/ART/ReactNativeART');

ReactNativeART.Group = rn.View;
ReactNativeART.Surface = rn.View;
ReactNativeART.Shape = rn.View;
module.exports = rn;
