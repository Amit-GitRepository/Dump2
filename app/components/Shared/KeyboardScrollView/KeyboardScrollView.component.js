import PropTypes from 'prop-types';
import React, {Component} from 'react';
import {KeyboardAwareScrollView} from 'react-native-keyboard-aware-scroll-view';

class KeyboardScrollView extends Component {
  render () {
    const {children, ...extraProps} = this.props;
    return (
      <KeyboardAwareScrollView  overScrollMode={'auto'} {...extraProps}>{children}</KeyboardAwareScrollView>
    );
  }
}

KeyboardScrollView.defaultProps = {
  children: null
};

KeyboardScrollView.propTypes = {
  children: PropTypes.node
};

export default KeyboardScrollView;
