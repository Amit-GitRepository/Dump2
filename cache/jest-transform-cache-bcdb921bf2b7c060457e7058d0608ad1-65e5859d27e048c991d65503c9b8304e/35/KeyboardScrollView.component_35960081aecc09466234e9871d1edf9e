Object.defineProperty(exports,"__esModule",{value:true});var cov_1qaly1sa1j=function(){var path='/Users/amitkumar/iTrueService/DevCode/iservice-app/app/components/Shared/KeyboardScollView/KeyboardScrollView.component.js',hash='5aaf5cc01e22eecbb71eebabe91ed60091e056c9',global=new Function('return this')(),gcv='__coverage__',coverageData={path:'/Users/amitkumar/iTrueService/DevCode/iservice-app/app/components/Shared/KeyboardScollView/KeyboardScrollView.component.js',statementMap:{'0':{start:{line:7,column:38},end:{line:7,column:48}},'1':{start:{line:8,column:4},end:{line:10,column:6}},'2':{start:{line:14,column:0},end:{line:16,column:2}},'3':{start:{line:18,column:0},end:{line:20,column:2}}},fnMap:{'0':{name:'(anonymous_0)',decl:{start:{line:6,column:2},end:{line:6,column:3}},loc:{start:{line:6,column:12},end:{line:11,column:3}},line:6}},branchMap:{},s:{'0':0,'1':0,'2':0,'3':0},f:{'0':0},b:{},_coverageSchema:'332fd63041d2c1bcb487cc26dd0d5f7d97098a6c'},coverage=global[gcv]||(global[gcv]={});if(coverage[path]&&coverage[path].hash===hash){return coverage[path];}coverageData.hash=hash;return coverage[path]=coverageData;}();var _extends=Object.assign||function(target){for(var i=1;i<arguments.length;i++){var source=arguments[i];for(var key in source){if(Object.prototype.hasOwnProperty.call(source,key)){target[key]=source[key];}}}return target;};var _createClass=function(){function defineProperties(target,props){for(var i=0;i<props.length;i++){var descriptor=props[i];descriptor.enumerable=descriptor.enumerable||false;descriptor.configurable=true;if("value"in descriptor)descriptor.writable=true;Object.defineProperty(target,descriptor.key,descriptor);}}return function(Constructor,protoProps,staticProps){if(protoProps)defineProperties(Constructor.prototype,protoProps);if(staticProps)defineProperties(Constructor,staticProps);return Constructor;};}();var _propTypes=require('prop-types');var _propTypes2=_interopRequireDefault(_propTypes);
var _react=require('react');var _react2=_interopRequireDefault(_react);
var _reactNativeKeyboardAwareScrollView=require('react-native-keyboard-aware-scroll-view');function _interopRequireDefault(obj){return obj&&obj.__esModule?obj:{default:obj};}function _objectWithoutProperties(obj,keys){var target={};for(var i in obj){if(keys.indexOf(i)>=0)continue;if(!Object.prototype.hasOwnProperty.call(obj,i))continue;target[i]=obj[i];}return target;}function _classCallCheck(instance,Constructor){if(!(instance instanceof Constructor)){throw new TypeError("Cannot call a class as a function");}}function _possibleConstructorReturn(self,call){if(!self){throw new ReferenceError("this hasn't been initialised - super() hasn't been called");}return call&&(typeof call==="object"||typeof call==="function")?call:self;}function _inherits(subClass,superClass){if(typeof superClass!=="function"&&superClass!==null){throw new TypeError("Super expression must either be null or a function, not "+typeof superClass);}subClass.prototype=Object.create(superClass&&superClass.prototype,{constructor:{value:subClass,enumerable:false,writable:true,configurable:true}});if(superClass)Object.setPrototypeOf?Object.setPrototypeOf(subClass,superClass):subClass.__proto__=superClass;}var

KeyboardScrollView=function(_Component){_inherits(KeyboardScrollView,_Component);function KeyboardScrollView(){_classCallCheck(this,KeyboardScrollView);return _possibleConstructorReturn(this,(KeyboardScrollView.__proto__||Object.getPrototypeOf(KeyboardScrollView)).apply(this,arguments));}_createClass(KeyboardScrollView,[{key:'render',value:function render()
{cov_1qaly1sa1j.f[0]++;var _ref=(cov_1qaly1sa1j.s[0]++,
this.props),children=_ref.children,extraProps=_objectWithoutProperties(_ref,['children']);cov_1qaly1sa1j.s[1]++;
return(
_react2.default.createElement(_reactNativeKeyboardAwareScrollView.KeyboardAwareScrollView,_extends({overScrollMode:'auto'},extraProps),children));

}}]);return KeyboardScrollView;}(_react.Component);cov_1qaly1sa1j.s[2]++;


KeyboardScrollView.defaultProps={
children:null};cov_1qaly1sa1j.s[3]++;


KeyboardScrollView.propTypes={
children:_propTypes2.default.node};exports.default=


KeyboardScrollView;