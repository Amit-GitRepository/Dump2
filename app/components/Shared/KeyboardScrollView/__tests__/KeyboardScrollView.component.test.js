import KeyboardScrollView from '../KeyboardScrollView.component';
import React from 'react';
import ReactSixteenAdapter from 'enzyme/build/adapters/ReactSixteenAdapter';
import renderer from 'react-test-renderer';
import {configure, shallow} from 'enzyme'; // Enzyme adapter for React 16
import {View} from 'react-native';

configure({adapter: new ReactSixteenAdapter()});

describe('KeyboardScrollView component', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<KeyboardScrollView><View /></KeyboardScrollView>);
    const component = renderer.create(<KeyboardScrollView><View /></KeyboardScrollView>).toJSON();
    expect(component).toMatchSnapshot();
    expect(wrapper).toBeDefined();
  });
});
