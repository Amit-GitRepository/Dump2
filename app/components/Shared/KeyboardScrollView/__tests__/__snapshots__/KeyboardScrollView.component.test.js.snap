// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`KeyboardScrollView component renders correctly 1`] = `
<RCTScrollView
  automaticallyAdjustContentInsets={false}
  contentContainerStyle={undefined}
  contentInset={
    Object {
      "bottom": 0,
    }
  }
  enableAutoAutomaticScroll={true}
  enableResetScrollToCoords={true}
  extraHeight={75}
  extraScrollHeight={0}
  keyboardDismissMode="interactive"
  keyboardOpeningTime={250}
  onScroll={[Function]}
  overScrollMode="auto"
  scrollEventThrottle={0}
  showsVerticalScrollIndicator={true}
>
  <View>
    <View />
  </View>
</RCTScrollView>
`;
